### PR TITLE
mapping body weight

### PR DESCRIPTION
**Pull request by MDSchulze**
Fixes ehrbase/num_platform_issues/45

### CODE REVIEW

File: src/main/java/org/ehrbase/fhirbridge/fhir/provider/ObservationResourceProvider.java [hunk 1]
```diff
@@ -498,10 +500,16 @@ public MethodOutcome createObservation(@ResourceParam Observation observation) {
...
+            } else if (ProfileUtils.hasProfile(observation, Profile.BODY_WEIGHT)) {
 
-            auditService.registerMapResourceEvent(AuditEvent.AuditEventOutcome._0, "Success", observation);
+                logger.info(">>>>>>>>>>>>>>>>>> OBSERVATION WEIGHT");
+
+                KorpergewichtComposition composition = FHIRObservationBodyWeightOpenehrBodyWeight.map(observation);
 
+                VersionUid versionUid = ehrbaseService.saveBodyWeight(ehrUid, composition);
+
+            auditService.registerMapResourceEvent(AuditEvent.AuditEventOutcome._0, "Success", observation);
```
Comment (ppazos) [inline src/main/java/org/ehrbase/fhirbridge/fhir/provider/ObservationResourceProvider.java:511]: @MDSchulze please fix indentation

File: src/main/java/org/ehrbase/fhirbridge/mapping/FHIRObservationBodyWeightOpenehrBodyWeight.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import ca.uhn.fhir.rest.server.exceptions.UnprocessableEntityException;
+
+public class FHIRObservationBodyWeightOpenehrBodyWeight {
+    private FHIRObservationBodyWeightOpenehrBodyWeight() {}
+
+    public static KorpergewichtComposition map(Observation fhirObservation) {
+
+        //create composition and observation objects
+    	KorpergewichtComposition composition = new KorpergewichtComposition();
```
Comment (ppazos) [inline src/main/java/org/ehrbase/fhirbridge/mapping/FHIRObservationBodyWeightOpenehrBodyWeight.java:25]: @MDSchulze please fix indentation

**Review by ppazos — CHANGES_REQUESTED** [review]
@MDSchulze please update the tests to pass https://app.circleci.com/pipelines/github/ehrbase/fhir-bridge/218/workflows/95835254-3ff6-4728-90f9-6cf2853fc865/jobs/222

**Comment by ppazos** [comment]
@MDSchulze can you please check the conflicts with develop branch and update the tests that are not passing?

**Comment by ppazos** [comment]
@SevKohler @MDSchulze I'm trying to test but getting:

`Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'observationResourceProvider' defined in URL [jar:file:/home/pab/GitHub/fhir-bridge/target/fhir-bridge-1.0.0-SNAPSHOT.jar!/org/ehrbase/fhirbridge/fhir/provider/ObservationResourceProvider.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'auditService' defined in URL [jar:file:/home/pab/GitHub/fhir-bridge/target/fhir-bridge-1.0.0-SNAPSHOT.jar!....`

Should I update something to get the SNAPSHOT dependency? Maybe @wlad knows about this?

**Comment by ppazos** [comment]
The issue above was because of a DB misconfiguration, now solved and the tests are passing.

**Review by ppazos — APPROVED** [review]
+1

**Comment by birgerhaarbrandt** [comment]
@ppazos please re-review with high priority, we need this by tomorrow

**Pull request merged**